dog_lover/src/breedContext.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useContext, useState } from "react";
+import axios from "axios";
+import {
+  DogBreed,
+  ListAllResponse,
+  ImageResponse,
+  DogBreedImage,
+} from "../components/breedSelectorBox";
+
+const ClickedBreedContext = React.createContext({} as DogBreedImage[]);
+const BreedContext = React.createContext({} as DogBreed[]);
+const HeartContextUpdate = React.createContext({} as any);
+const FetchBreedsContextUpdate = React.createContext({} as any);
+const OpenContext = React.createContext({} as boolean);
+const QueryContext = React.createContext({} as any);
+const QueryContextUpdate = React.createContext({} as any);
+
+export const useVariableBreedContext = () => {
+  return useContext(BreedContext);
+};
+export const useVariableClickedBreedContext = () => {
+  return useContext(ClickedBreedContext);
+};
+export const useVariableOpenContext = () => {
+  return useContext(OpenContext);
+};
+export const useVariableQueryContext = () => {
+  return useContext(QueryContext);
+};
+export const useFunctionHeartContext = () => {
+  return useContext(HeartContextUpdate);
+};
+export const useFunctionFetchBreedsContext = () => {
+  return useContext(FetchBreedsContextUpdate);
+};
+export const useFunctionQueryContext = () => {
+  return useContext(QueryContextUpdate);
+};
+
+export const BreedProvider = ({ children }: { children: any }) => {
+  const [breeds, setBreeds] = useState<DogBreed[]>([]);
+  const [clickedBreed, setClickedBreed] = useState<DogBreedImage[]>([]);
+  const [isOpened, setIsOpened] = useState(false);
+  const [query, setQuery] = useState("");
+
+  const toggle = () => {
+    setIsOpened((wasOpened) => !wasOpened);
+    console.log("toggle");
+  };
+
+  const handleHeartClicked = (breed: string) => () => {
+    setClickedBreed((state) => ({
+      ...state, // <-- copy previous state
+      [breed]: !state[breed], // <-- update value by index key
+    }));
+    toggle();
+  };
+
+  const fetchBreedsList = async () => {
+    // fetch all breeds. Axios requests can be typed for the response
+    const {
+      data: { message: allBreeds },
+    } = await axios.get<ListAllResponse>("https://dog.ceo/api/breeds/list/all");
+
+    // extract keys from the ListAllResponse.message
+    const keys = Object.keys(allBreeds);
+
+    // resolve images and create DogBreed objects
+    const dogBreeds = await Promise.all(
```

Comment:
What will happen if one of the promises fail?

Reply:
You are absolutely right, I thought I could get away with not adding a try/catch! Will do it in my next commit if I am allowed to of course.